sway/commands/input/xkb_switch_layout.c

```diff
@@ -99,6 +99,11 @@ struct cmd_results *input_cmd_xkb_switch_layout(int argc, char **argv) {
 			&actions[actions_len++];
 
 		action->keyboard = wlr_keyboard_from_input_device(dev->wlr_device);
+		if (action->keyboard->keymap == NULL && dev->is_virtual) {
+			actions_len -= 1;
```

Comment:
Instead of incrementing right above and decrementing after-the-fact, can we store the result of `wlr_keyboard_from_input_device()` in an intermediate variable and `continue` if it's NULL?